main/src/main/scala/sbt/Cross.scala

```diff
@@ -361,10 +361,15 @@ object Cross {
     }
 
     if (included.isEmpty) {
-      sys.error(
-        s"""Switch failed: no subprojects list "$version" (or compatible version) in crossScalaVersions setting.
-           |If you want to force it regardless, call ++ $version!""".stripMargin
-      )
+      if (isSelector(version))
+        sys.error(
```

Comment:
The change itself looks good. Since this is a somewhat predictable validation error, so could we throw `sbt.internal.util.MessageOnlyException` here and below please? That will show the error without a stacktrace.